websites/V/Vizzy/presence.ts

```diff
@@ -0,0 +1,127 @@
+const presence = new Presence({
+		clientId: "968353669491871754"
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+function getElementByXpath(path: string) {
+	return document.evaluate(
+		path,
+		document,
+		null,
+		XPathResult.FIRST_ORDERED_NODE_TYPE,
+		null
+	).singleNodeValue;
+}
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			smallImageKey: "logo_512",
+			startTimestamp: browsingTimestamp
+		},
+		{ pathname, search, href } = window.location;
+	switch (pathname) {
+		case "/": {
+			presenceData.details = "Viewing the homepage";
+			presenceData.largeImageKey = "home";
+			break;
+		}
+		case "/creations":
+		case "/community": {
+			presenceData.details = "Viewing community creations";
+			presenceData.largeImageKey = "discover";
+			break;
+		}
+		case "/project": {
+			presenceData.largeImageKey = "discover";
+			presenceData.details = "Viewing a project";
+			presenceData.state = getElementByXpath(
+				"/html/body/div[1]/div/main/div/div/div[2]/div/div[2]/div[1]/h4"
+			).textContent;
+			presenceData.buttons = [
+				{
+					label: "View Project",
+					url: href
+				}
+			];
+			break;
+		}
+		case "/user": {
+			presenceData.largeImageKey = "discover";
+			presenceData.details = "Viewing a user's page";
+			presenceData.state = `@${
+				getElementByXpath("/html/body/div[1]/div/h5").textContent
+			}`;
+			presenceData.buttons = [
+				{
+					label: "View User",
+					url: href
+				}
+			];
+			break;
+		}
+		case "/profile": {
+			presenceData.details = "Viewing their profile";
+			presenceData.largeImageKey = "profile";
+			if (search) {
+				presenceData.state = `Tab: ${new URLSearchParams(
+					document.location.search
+				).get("tab")}`;
+			}
+			break;
+		}
+		case "/editor": {
+			if (
+				getElementByXpath(
+					"/html/body/div[1]/div[2]/header/div[4]/div[3]/span[1]"
+				).textContent === "Community project"
+			)
+				presenceData.details = "Editing a community project";
+			else presenceData.details = "Editing a project";
+			presenceData.state = `Project: ${
+				getElementByXpath(
+					"/html/body/div[1]/div[2]/header/div[4]/div[1]/span/span"
+				).textContent
+			}`;
+			presenceData.largeImageKey = "editor";
+			break;
+		}
+		case "/export": {
+			presenceData.largeImageKey = "export";
+			if (getElementByXpath("/html/body/div[1]/div/div/div[2]/div[2]/p")) {
+				presenceData.details = "Exporting a video";
+				presenceData.state = getElementByXpath(
+					"/html/body/div[1]/div/div/div[2]/div[2]/p"
+				).textContent;
+			} else presenceData.details = "Preparing to export a video";
+			break;
+		}
+		case "/discover": {
+			const tabParams = new URLSearchParams(search).get("tab");
```

Comment:
I believe I suggested making a searchParams variable at the top so you don't have to compute this so often